testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindFixtures.scala

```diff
@@ -0,0 +1,318 @@
+    extends BitcoindFixturesCached
+    with CachedBitcoindPair[T] {
+  _: BitcoinSAsyncFixtureTest =>
+
+  def with2BitcoindsCached(
+      test: OneArgAsyncTest,
+      bitcoinds: NodePair[T]): FutureOutcome = {
+    makeDependentFixture[NodePair[T]](
+      () => Future.successful(bitcoinds),
+      destroy = { case _: NodePair[T] =>
+        //do nothing since we are caching bitcoinds
+        //the test trait may want to re-use them
+        Future.unit
+      }
+    )(test)
+  }
+}
+
+/** Bitcoind fixtures with two cached BitcoindV16RpcClient that are connected via p2p */
+trait BitcoindFixturesCachedPairV16
+    extends BitcoinSAsyncFixtureTest
+    with BitcoindFixturesCachedPair[BitcoindV16RpcClient] {
+  override type FixtureParam = NodePair[BitcoindV16RpcClient]
+
+  override val version: BitcoindVersion = BitcoindVersion.V16
+
+  override def withFixture(test: OneArgAsyncTest): FutureOutcome = {
+    val futOutcome = for {
+      pair <- clientsF
+      futOutcome = with2BitcoindsCached(test, pair)
+      f <- futOutcome.toFuture
+    } yield f
+    new FutureOutcome(futOutcome)
+  }
+
+  override def afterAll(): Unit = {
+    super[BitcoindFixturesCachedPair].afterAll()
+    super[BitcoinSAsyncFixtureTest].afterAll()
+  }
+}
+
+/** Bitcoind fixtures with two cached [[BitcoindV17RpcClient]] that are connected via p2p */
+trait BitcoindFixturesCachedPairV17
+    extends BitcoinSAsyncFixtureTest
+    with BitcoindFixturesCachedPair[BitcoindV17RpcClient] {
+  override type FixtureParam = NodePair[BitcoindV17RpcClient]
+
+  override val version: BitcoindVersion = BitcoindVersion.V17
+
+  override def withFixture(test: OneArgAsyncTest): FutureOutcome = {
+    val futOutcome = for {
+      pair <- clientsF
+      futOutcome = with2BitcoindsCached(test, pair)
+      f <- futOutcome.toFuture
+    } yield f
+    new FutureOutcome(futOutcome)
+  }
+
+  override def afterAll(): Unit = {
+    super[BitcoindFixturesCachedPair].afterAll()
+    super[BitcoinSAsyncFixtureTest].afterAll()
+  }
+}
+
+/** Bitcoind fixtures with two cached [[BitcoindV18RpcClient]] that are connected via p2p */
+trait BitcoindFixturesCachedPairV18
+    extends BitcoinSAsyncFixtureTest
+    with BitcoindFixturesCachedPair[BitcoindV18RpcClient] {
+  override type FixtureParam = NodePair[BitcoindV18RpcClient]
+
+  override val version: BitcoindVersion = BitcoindVersion.V18
+
+  override def withFixture(test: OneArgAsyncTest): FutureOutcome = {
+    val futOutcome = for {
+      pair <- clientsF
+      futOutcome = with2BitcoindsCached(test, pair)
+      f <- futOutcome.toFuture
+    } yield f
+    new FutureOutcome(futOutcome)
+  }
+
+  override def afterAll(): Unit = {
+    super[BitcoindFixturesCachedPair].afterAll()
+    super[BitcoinSAsyncFixtureTest].afterAll()
+  }
+}
+
+/** Bitcoind fixtures with two cached bitcoind rpc clients that are [[BitcoindVersion.newest]] that are connected via p2p */
+trait BitcoindFixturesCachedPairNewest
```

Comment:
Unfortunately that is the name of the parent trait that is extended by all versions. I would have to come up with a different name for that (any suggestions?)

Reply:
Maybe just change `newest` to `lastest` that sounds better imo

Reply:
The reason I used `newest` is because we have the field named that 

https://github.com/bitcoin-s/bitcoin-s/blob/89185338e04b413d64cfd3175a917e816601246a/bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala#L280

I can still change if you want, but i thought i would provide the context